### PR TITLE
article preview and pagination

### PR DESCRIPTION
**Pull request by jkao1**
Generate article previews on article creation.

### CODE REVIEW

File: app/models/article.rb [hunk 1]
```diff
@@ -1,27 +1,46 @@
...
-
   has_many :media, dependent: :destroy
   has_many :comments, dependent: :destroy
   has_many :outquotes, dependent: :destroy
 
   def init
     self.update(is_published: false)
+
+    if self.summary.to_s.empty?
+        preview = self.content.split(' ')[0, 25].join(' ') + '...'
```
Comment (NicholasLYang) [inline app/models/article.rb]: What's with the to_s?

File: app/models/article.rb [hunk 2]
```diff
@@ -1,27 +1,46 @@
...
+      if words.length > 25
+        preview = words[0, 25].join(' ') + '...'
+      else
+        preview = self.summary
+      end
+    end
+
+    # Replacing </p><p> is separate from the regex sub so we don't get two
+    # spaces in a row.
+    self.update(preview, preview.gsub('</p><p>', ' ').gsub(/<\/?[^>]*>/, ' '))
```
Comment (NicholasLYang) [inline app/models/article.rb]: What does the second gsub do?

Comment (jkao1) [reply app/models/article.rb]: Global replace

File: app/models/article.rb (hunk 1 above)
Comment (jkao1) [reply app/models/article.rb]: `to_s` is to string. If the summary is nil, it will became an empty string.

Comment (NicholasLYang) [reply app/models/article.rb]: why not self.summary.nil?

File: app/models/article.rb (hunk 2 above)
Comment (NicholasLYang) [reply app/models/article.rb]: Of what? I know what gsub is. I'm asking about the regex

File: app/models/article.rb (hunk 1 above)
Comment (jkao1) [reply app/models/article.rb]: Just simpler/easier to do it one line. Why have two cases when you can cover both in one command?

File: app/models/article.rb (hunk 2 above)
Comment (jkao1) [reply app/models/article.rb]: Sorry didn't read your comment clearly. It replaces all HTML tags. I'll add a comment.

File: app/models/article.rb (hunk 1 above)
Comment (NicholasLYang) [reply app/models/article.rb]: No, like why are you converting to int then checking if it's empty when you can just check when it's nil

Comment (jkao1) [reply app/models/article.rb]: Fine. Your solution is more readable.

File: app/models/article.rb [hunk 3]
```diff
@@ -1,30 +1,49 @@
...
            source: :user
-
   has_many :media, dependent: :destroy
   has_many :comments, dependent: :destroy
   has_many :outquotes, dependent: :destroy
 
   def init
     self.update(is_published: false)
+
+    if self.summary.nil? or self.summary.summary.empty?
```
Comment (NicholasLYang) [inline app/models/article.rb]: This is fine here, but be very careful about `or` vs `||` in Ruby

Comment (NicholasLYang) [reply app/models/article.rb]: https://stackoverflow.com/questions/2083112/difference-between-or-and-in-ruby

Comment (jkao1) [reply app/models/article.rb]: Thanks for the reference.

File: app/graphql/resolvers/get_latest_articles.rb [hunk 4]
```diff
@@ -2,16 +2,20 @@ class Resolvers::GetLatestArticles < GraphQL::Function
...
-   # _obj - is parent object, which in this case is nil
+  # _obj - is parent object, which in this case is nil
   # _args - are the arguments passed
   # _ctx - is the GraphQL context (which would be discussed later)
   def call(_obj, args, _ctx)
     articles = Article.order(created_at: :desc)
-    articles = articles.where(section_id: args["section_id"]) if args["section_id"]
-    return articles.limit(args["limit"]) if args["limit"]
+    articles = articles.where(section_id: args['section_id']) if args['section_id']
+    articles = articles.limit(args['limit']) if args['limit']
```
Comment (NicholasLYang) [inline app/graphql/resolvers/get_latest_articles.rb]: Do we still need limit?

File: app/models/article.rb [hunk 5]
```diff
@@ -1,30 +1,49 @@
...
+      if words.length > 25
+        preview = words[0, 25].join(' ') + '...'
+      else
+        preview = self.summary
+      end
+    end
+
+    # Removes all HTML tags. We replace </p><p> separately so there are never
+    # two spaces in a row.
+    self.update(preview, preview.gsub('</p><p>', ' ').gsub(/<\/?[^>]*>/, ' '))
```
Comment (NicholasLYang) [inline app/models/article.rb]: You sure this will work? How about <div class="hello >">

Comment (NicholasLYang) [reply app/models/article.rb]: \<div id="hello \>"\>

Comment (NicholasLYang) [reply app/models/article.rb]: https://stackoverflow.com/questions/7414267/strip-html-from-string-ruby-on-rails

**Comment by jkao1** [comment]
Closed in favor of #92.

**Pull request closed without merging**